components/eamxx/src/physics/p3/tests/CMakeLists.txt

```diff
@@ -61,36 +61,57 @@ if (NOT SCREAM_BASELINES_ONLY)
 endif()
 
 if (SCREAM_ENABLE_BASELINE_TESTS)
-  CreateUnitTest(p3_run_and_cmp_cxx "p3_run_and_cmp.cpp" "${NEED_LIBS}"
+  set(BASELINE_FILE_ARG "-b ${SCREAM_TEST_DATA_DIR}/p3_run_and_cmp.baseline")
+
+  CreateUnitTestExec(p3_run_and_cmp "p3_run_and_cmp.cpp" "${NEED_LIBS}"
+                     EXCLUDE_MAIN_CPP)
+
+  CreateUnitTestFromExec(p3_run_and_cmp_cxx p3_run_and_cmp
                  THREADS ${SCREAM_TEST_MAX_THREADS}
-                 EXE_ARGS "-b ${SCREAM_TEST_DATA_DIR}/p3_run_and_cmp.baseline"
-                 EXCLUDE_MAIN_CPP
+                 EXE_ARGS "${BASELINE_FILE_ARG}"
                  LABELS "p3;physics")
 
-  CreateUnitTest(p3_run_and_cmp_f90 "p3_run_and_cmp.cpp" "${NEED_LIBS}"
+  CreateUnitTestFromExec(p3_run_and_cmp_f90 p3_run_and_cmp
                  THREADS ${SCREAM_TEST_MAX_THREADS}
-                 EXE_ARGS "-f -b ${SCREAM_TEST_DATA_DIR}/p3_run_and_cmp.baseline"
-                 EXCLUDE_MAIN_CPP
+                 EXE_ARGS "-f ${BASELINE_FILE_ARG}"
                  LABELS "p3;physics")
 
   # Make sure that a diff from baselines triggers a failed test (in debug only)
   CreateUnitTest(p3_run_and_cmp_cxx_fail "p3_run_and_cmp.cpp" "${NEED_LIBS}"
                  COMPILER_CXX_DEFS SCREAM_FORCE_RUN_DIFF
                  THREADS ${SCREAM_TEST_MAX_THREADS}
-                 EXE_ARGS "-b ${SCREAM_TEST_DATA_DIR}/p3_run_and_cmp.baseline"
+                 EXE_ARGS "${BASELINE_FILE_ARG}"
                  PROPERTIES WILL_FAIL ${FORCE_RUN_DIFF_FAILS}
                  EXCLUDE_MAIN_CPP
                  LABELS "p3;physics;fail")
 
-  # By default, baselines should be created using all fortran (make baseline). If the user wants
-  # to use CXX to generate their baselines, they should use "make baseline_cxx".
+  #
+  # Use fake tests to generate shell commands to generate baselines
+  #
+  CreateUnitTestFromExec(p3_baseline_f90_fake p3_run_and_cmp
+    THREADS ${SCREAM_TEST_MAX_THREADS}
+    EXE_ARGS "-f -g ${BASELINE_FILE_ARG}"
+    PROPERTIES DISABLED True)
+
+  CreateUnitTestFromExec(p3_baseline_cxx_fake p3_run_and_cmp
+    THREADS ${SCREAM_TEST_MAX_THREADS}
+    EXE_ARGS "-g ${BASELINE_FILE_ARG}"
+    PROPERTIES DISABLED True)
+
+  get_test_property(p3_baseline_f90_fake_omp${SCREAM_TEST_MAX_THREADS} FULL_TEST_COMMAND P3_F90_GEN)
+  get_test_property(p3_baseline_cxx_fake_omp${SCREAM_TEST_MAX_THREADS} FULL_TEST_COMMAND P3_CXX_GEN)
+
+  separate_arguments(P3_F90_GEN_ARGS UNIX_COMMAND "${P3_F90_GEN}")
+  separate_arguments(P3_CXX_GEN_ARGS UNIX_COMMAND "${P3_CXX_GEN}")
 
   add_custom_target(p3_baseline_f90
-    COMMAND ${CMAKE_COMMAND} -E env OMP_NUM_THREADS=${SCREAM_TEST_MAX_THREADS} $<TARGET_FILE:p3_run_and_cmp_f90> -f -g -b ${SCREAM_TEST_DATA_DIR}/p3_run_and_cmp.baseline)
+    COMMAND ${CMAKE_COMMAND} -E env OMP_NUM_THREADS=${SCREAM_TEST_MAX_THREADS} ${P3_F90_GEN_ARGS})
```

Comment:
I'm puzzled. Doesn't this run the same command that the original version would?

Reply:
No, ECUT adds a bunch of stuff like test-launcher, mpi stuff, etc.

Reply:
Ah, so batch machines will run the generation through the queue. Ok, thanks.